sr_wear_logger/src/sr_wear_logger.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+#
+# Copyright <Year> Shadow Robot Company Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+import rospy
+import rospkg
+import os
+import yaml
+import roslaunch
+import requests
+import shutil
+from sensor_msgs.msg import JointState
+from sr_utilities_common.awsmanager import AWS_Manager
+
+
+class WearLogger():
+
+    def _extractHandData(self, msg):
+        hand_data = dict(zip(msg.name, msg.position))
+        for key in hand_data.keys():
+            if key.startswith("ra") or key.startswith("la"):
+                hand_data.pop(key, None)
+        return hand_data
+
+    def callback(self, msg):
+        hand_data = self._extractHandData(msg)
+        updates = dict.fromkeys(hand_data.keys(), 0)
+
+        if self.first_run:
+            self.previousValues = hand_data
+            self.first_run = False
+
+        for key, value in hand_data.items():
+            valueDifference = abs(self.previousValues[key] - value)
+            if valueDifference > self.threshold:
+                self.currentValues[key] += round(valueDifference, 5)
+        self.previousValues = hand_data
+
+    def _uploadToAWS(self, event):
+        self.aws_manager.upload("shadowrobot.benchmarks", rospkg.RosPack().get_path('sr_wear_logger'),
+                                "data", ["wear_data.yaml"])
+        return rospy.get_param('aws_upload_succeeded')
+
+    def _downloadFromAWS(self):
+        self.aws_manager.download("shadowrobot.benchmarks", rospkg.RosPack().get_path('sr_wear_logger'),
+                                  "data", ["wear_data.yaml"])
+        return rospy.get_param('aws_download_succeeded')
+
+    def _saveDataLocaly(self, event):
+        if self._verifyData():
+            f = open(self.logFilePath+self.logFileName, 'w')
+            completeData = dict()
+            completeData['total_angles'] = self.currentValues
+            completeData['total_time'] = rospy.get_rostime().secs
+            yaml.safe_dump(completeData, f)
+            f.close()
+
+    def __init__(self):
+        self.aws_manager = AWS_Manager()
+        self.first_run = True
+        self.previousValues = {}
+        self.currentValues = {}
+        self.threshold = 0.0175
+        self.logFilePath = "../data/"
```

Comment:
we can use the rospack find API to make sure we always find the right path, and would be easier to change in future. here example: https://github.com/shadow-robot/sr_demos/blob/ca2efff8136b0a25cfbd5b1ece02a5e235c4cca6/sr_lab_demo/src/sr_lab_demo/lab_setup.py#L74-L75

Reply:
Corrected